save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt

```diff
@@ -179,26 +212,44 @@ class DemoManagerController(
         description = "Get demo info.",
     )
     @ApiResponse(responseCode = "200", description = "Successfully fetched demo status.")
+    @ApiResponse(responseCode = "403", description = "Not enough permission for accessing given project.")
+    @ApiResponse(responseCode = "404", description = "Could not find project in organization.")
     fun getDemoInfo(
         @PathVariable organizationName: String,
         @PathVariable projectName: String,
         authentication: Authentication,
-    ): Mono<DemoInfo> = webClientDemo.get()
-        .uri("/demo/internal/$organizationName/$projectName")
-        .retrieve()
-        .onStatus({ !it.is2xxSuccessful }) {
-            Mono.error(
-                ResponseStatusException(
-                    HttpStatus.NOT_FOUND,
-                    "Demo for $organizationName/$projectName is not found.",
-                )
-            )
+    ): Mono<DemoInfo> = blockingToMono {
+        projectService.findByNameAndOrganizationNameAndCreatedStatus(projectName, organizationName)
+    }
+        .switchIfEmptyToNotFound {
+            "Could not find project $projectName in organization $organizationName."
         }
-        .bodyToMono<DemoInfo>()
-        .defaultIfEmpty(
-            DemoInfo(
-                DemoDto.emptyForProject(organizationName, projectName),
-                DemoStatus.NOT_CREATED,
+        .filter {
+            projectPermissionEvaluator.hasPermission(
+                authentication, it, Permission.WRITE
```

Comment:
why WRITE for GET?

Reply:
In the issue #1676 @sanyavertolet wrote on which permissions to check

Reply:
@nulls This GET method might return some internal information that should not be available for everyone (in case of public project)